pkg/retry/retry.go

```diff
@@ -9,6 +9,10 @@ const (
 	Interval = 15 * time.Second
 	// Timeout is the total time we will wait for an event to occur.
 	Timeout = time.Minute * 10
+	// TCPConnTimeout is the maximum amount of time for a single TCP connection to establish
+	TCPConnTimeout = time.Second * 45
+	// SSHRetryTimeout is the total time we will retry to establish an SSH connection with a particular socket address
+	SSHRetryTimeout = time.Minute * 5
```

Comment:
The Windows networking stack takes a while to get configured especially on 1st boot. Are you certain that reducing this to 5 minutes won't introduce flakes that we might miss? i.e. the instance will get configured on the 2nd try.

Reply:
I personally don't feel like changing this timeout logic without making sure we dont regress ( by creating any new flakyness)

Reply:
I've tested this on different platforms and did not see any issue (tested 5 times total). Though I've also looked through some recent CI jobs in different PRs and have not seen any where this logic would cause a premature reconcile error.

@aravindhp @selansen Aside from this manual testing, is there recommended way to ensure we wouldn't regress?

Reply:
We might have to test this out in CI as well, before we can ensure this doesn't introduce flakes